### ISSUE TITLE
On Xamarin iOS gets stuck on Realm.GetInstanceAsync call when using a path with an identity like "realms://{SyncHost}/{identity}/{name}" in Realm Cloud Platform

### ISSUE THREAD

**Issue by rdelrosario** [realm/realm-dotnet#1733]
When using Realm Platform Cloud cannot get the realm instance, task never returns a result when calling Realm.GetInstanceAsync for a private shared user realm with path like "realms://{Server.SyncHost}/{identity}/{name}".

```cs
await Realm.GetInstanceAsync(new SyncConfiguration(User.Current, new Uri($"realms://{Server.SyncHost}/{identity}/{name}")){
                EnableSSLValidation = false,
		ObjectClasses = new Type[]{ typeof(Models.Message), typeof(Models.Avatar), typeof(Models.Attachment) }
            });
```

## Goals

Get the realm database instance when using a realm with an identity path.

## Expected Results

Should return the Realm instance as when using **Realm.GetInstanceAsync**  like when gets instance for a public realm but for private shared realms not working.

## Actual Results

When using **Realm.GetInstanceAsync** method to get a private shared Realm with an identity path never gets a response. For public realms is working correctly.

Working with:
Path: "realms://{SyncHost}/{name}"

Not working with:
Path: "realms://{SyncHost}/{identity}/{name}"

## Steps to Reproduce

1. Create a realm platform cloud instance
2. Create a reference type realm path with some classes inside.
3. Create a Realm User 
4. Try to login with username/password
5. Try to get  realm by using **Realm.GetInstanceAsync** and a Realm path with an identity like:

Example: "realms://{SyncHost}/088e246e292519eb885a82eee8c3b36f/example"

## Code Sample

```cs
await Realm.GetInstanceAsync(new SyncConfiguration(User.Current, new Uri($"realms://{Server.SyncHost}/{identity}/{name}")){
                EnableSSLValidation = false,
		ObjectClasses = new Type[]{ typeof(Models.Message), typeof(Models.Avatar), typeof(Models.Attachment) }
            });
```

## Version of Realm and Tooling

- Realm Cloud

- Flavor:
  - [X] Developer
  - [ ] Professional
  - [ ] Enterprise

- Client SDK Version:  3.0.0
- Client OS & Version: iOS 11.0

**Comment by nirinchev** [comment]
Can you wrap the `GetInstanceAsync` call in try-catch and check if an exception is thrown? While our tests aren't perfect, we have quite a few doing that and they haven't failed recently.

**Comment by rdelrosario** [comment]
Even on a try-catch block, no exception is thrown just gets stuck there forever.

**Comment by rdelrosario** [comment]
Not happening after deleting my instance and creating a new one.

**Comment by nirinchev** [comment]
A reference Realm cannot be opened in full-sync mode by non-admin users. I imagine there was a session error that reported an authentication failure. We have plans to improve the error reporting there so you don't need to write error handling code in two different places, but that hasn't been prioritized yet. For future reference, if you need to verify why things are not synchronizing (as it is a prerequisite for `GetInstanceAsync`) you can attach an error handler on the `Session` class:

```csharp
Session.Error += (session, error) =>
{
    Console.WriteLine(error);
};
```

**Comment by tekr** [comment]
I have been having almost the same problem for the last day using Xamarin with some XUnit tests.

1. Signed up for a trial Realm Cloud account. The email verification step kept failing with 'Invalid token' from the Realm cloud website, and after contacting Realm support, they fixed it somehow (may be relevant, maybe not). Created a ROS instance (success).

2. Wrote some simple unit tests to trial cloud sync. First, create a user using username/password (success), then log in (success). However calling GetInstanceAsync() would hang indefinitely, with no exceptions, and no errors reported from the Session.Error event, just like @rdelrosario mentioned above.

3. GetInstance() worked however, and I was able to proceed and create the private Realm (using a '~' in the path as per the docs) and put a couple of items in it. The local Realm seemed to persist the data fine (it was there after restart), however it was never synced to the cloud Realm. I verified in Realm Studio that the private Realm was created on the server when I ran the test, and that the realm session state was 'Active', however the cloud Realm remained empty with no class metadata, and there were no items published on the sync progress upload/download observables. Note the Realm type was 'Full', not 'Reference' as mentioned above.

4. Deleting the local and server private Realm and letting the test re-create it or using a different user did not seem to make any difference.

5. What did help? I signed up for a completely new Realm Cloud trial account (thanks @rdelrosario for mentioning this above!). Created a new ROS, repeated the same steps above, and everything worked just fine! GetInstanceAsync() returns, cloud sync works, and I get items published to the sync progress observables.
